skglm/utils.py

```diff
@@ -254,3 +255,46 @@ def check_group_compatible(obj):
                 f"'{obj_name}' is not block-separable. "
                 f"Missing '{attr}' attribute."
             )
+
+
+spec_AndersonAcceleration = [
+    ('K', int32),
+    ('n_features', int32),
+    ('current_iter', int32),
+    ('arr_w', float64[:, ::1]),
+]
+
+
+@jitclass(spec_AndersonAcceleration)
+class AndersonAcceleration:
+    """Anderson Accelerator."""
+
+    def __init__(self, K, n_features):
+        self.K, self.n_features = K, n_features
+
+        self.current_iter = 0
+        self.arr_w = np.zeros(shape=(self.n_features, self.K+1))
+
+    def extrapolate(self, w):
+        """Inplace update of ``w``."""
+        K, current_iter = self.K, self.current_iter
+        arr_w = self.arr_w
+        n_features = self.n_features
+
+        if current_iter <= K:
+            self.arr_w[:, current_iter] = w.copy()
+            self.current_iter += 1
+            return
+
+        # compute residuals
+        U = np.zeros((n_features, K))
+        for j in range(K):
+            U[:, j] = arr_w[:, j+1] - arr_w[:, j]
+
+        # compte extrapolation coefs
+        ones_K = np.ones(K)
+        inv_UTU_ones = np.linalg.lstsq(U.T @ U, ones_K)[0]
```

Comment:
@mathurinm, @PABannier
Any idea on handling the case of a singular matrix here (without try: except)

I am not that convinced about using a least-square solution

Reply:
Check how we do it in `cd_solver`. Wrap the call to `np.linalg.inv` in a `try`- `except` block and catch `LinAlgError`.

Reply:
see: https://github.com/scikit-learn-contrib/skglm/blob/main/skglm/solvers/cd_solver.py#L300